openstudiocore/src/utilities/filetypes/EpwFile.cpp

```diff
@@ -748,6 +982,223 @@ namespace openstudio{
     return string;
   }
 
+  boost::optional<std::string> EpwDesignCondition::getUnitsByName(const std::string &name)
+  {
+    EpwDesignField id;
+    try {
+      id = EpwDesignField(name);
+    }
+    catch (...) {
+      // Could do a warning message here
+      return boost::none;
+    }
+    return boost::optional<std::string>(getUnits(id));
```

Comment:
You can just return getUnits(id).

Reply:
@macumber I'm confused why this wouldn't be an optional. Why doesn't EpwDataPoint::getUnitsByName(...) return an optional then too?

Reply:
The result is an optional, you just don't need to explicitly construct one like this.  If you return a string, the compiler can implicitly convert that to an optional string.